public/app/features/dashboard-scene/scene/DashboardDatasourceBehaviour.tsx

```diff
@@ -58,6 +82,25 @@ export class DashboardDatasourceBehaviour extends SceneObjectBase<DashboardDatas
 
     return () => {
       this.prevRequestId = sourcePanelQueryRunner?.state.data?.request?.requestId;
+      if (libraryPanelSub) {
+        libraryPanelSub.unsubscribe();
+      }
     };
   }
+
+  private handleLibPanelStateUpdates(
+    newLibPanel: LibraryPanelBehaviorState,
+    oldLibPanel: LibraryPanelBehaviorState,
+    queryRunner: SceneQueryRunner,
+    sourcePanel: VizPanel
+  ) {
+    if (newLibPanel && newLibPanel?.isLoaded && newLibPanel !== oldLibPanel) {
```

Comment:
I think we don't even need the oldPanel here, just checking if `isLoaded` is true should guarantee that we have proper data